conan/tools/gnu/autotools.py

```diff
@@ -75,36 +65,33 @@ def _fix_install_name(lib_name, lib_folder):
                                                                                          lib_name))
             self._conanfile.run(command)
 
-        def _is_modified_install_name(lib_name, lib_folder):
+        def _is_modified_install_name(lib_name, full_folder, libdir):
             """
             Check that the user did not change the default install_name using the install_name
             linker flag in that case we do not touch this field
             """
-            command = "otool -D {}".format(os.path.join(lib_folder, lib_name))
-            out = check_output_runner(command).strip().split(":")[1]
-            return False if str(os.path.join(lib_folder, shared_lib)) in out else True
+            command = "otool -D {}".format(os.path.join(full_folder, lib_name))
+            install_path = check_output_runner(command).strip().split(":")[1].strip()
+            default_path = str(os.path.join("/", libdir, shared_lib))
+            return False if default_path == install_path else True
 
         libdirs = getattr(self._conanfile.cpp.package, "libdirs")
-        for folder in libdirs:
-            full_folder = os.path.join(self._conanfile.package_folder, folder)
+        for libdir in libdirs:
+            full_folder = os.path.join(self._conanfile.package_folder, libdir)
             shared_libs = _osx_collect_dylibs(full_folder)
             for shared_lib in shared_libs:
-                if not _is_modified_install_name(shared_lib, full_folder):
+                if not _is_modified_install_name(shared_lib, full_folder, libdir):
                     _fix_install_name(shared_lib, full_folder)
 
-    def install(self):
-        # FIXME: we have to run configure twice because the local flow won't work otherwise
-        #  because there's no package_folder until the package step
-        self.configure()
-        self.make(target="install")
+    def install(self, args=None):
+        args = args if args is not None else ["DESTDIR={}".format(self._conanfile.package_folder)]
+        self.make(target="install", args=args)
         if self._conanfile.settings.get_safe("os") == "Macos" and self._conanfile.options.get_safe("shared", False):
             self._fix_osx_shared_install_name()
 
     def autoreconf(self, args=None):
-        command = ["autoreconf"]
-        args = args or ["--force", "--install"]
-        command.extend(args)
-        command = join_arguments(command)
+        args = args or []
```

Comment:
Whilst getting familiar with the Conan code base, I noticed this Python anti-pattern.

The code base could simplify the code flow by moving towards immutable tuples rather than mutable lists.

```py
# Python3 with expandable iterables syntax
def autoreconf(self, args=("--force", "--install")):
    command = join_arguments(("autoreconf", self._autoreconf_args, *args))
    with chdir(self, self._conanfile.source_folder):
        self._conanfile.run(command)

# Python2/3 with extending from iterables
def autoreconf(self, args=("--force", "--install")):
    command = ["autoreconf"]
    command.extend(self._autoreconf_args)
    command.extend(args)
    with chdir(self, self._conanfile.source_folder):
        self._conanfile.run(command)
```

The above is totally valid because the static binding of the tuple to the `args` is immutable it is not possible to change the bound value within the function. This is a common problem in code bases that use mutable bindings in argument defaults which we are avoiding by using `None`. i.e. the following would break terribly:

```py
def autoreconf(self, args=["--force", "--install"]):
    args.append("foobar")  # everytime `autoreconf` is invoked `args` gets longer
```

In general, I personally use immutable values as much as possible to avoid unintended side effects. It's a shame that Python doesn't have records for immutable dictionaries.